### DISCUSSION TITLE
Error in using Contact properties with Anura3D_2022

### DISCUSSION THREAD

**Discussion by Tiancheng-Xie** [Anura3D/Anura3D_OpenSource#72] [Q&A] [answered]
Hello, everyone! When I simulate the shallow foundation example in Tutorial Manual 2022, I want to add a contact between the foundation and soil. However, the calculation is always error, and I don't know why. Please see the picture.
![image](https://user-images.githubusercontent.com/106669043/222163519-bc697b60-ef42-48c2-ac24-55cbefefaf25.png)
![image](https://user-images.githubusercontent.com/106669043/222163598-091f8f99-0de4-4cf3-b916-a2960723e946.png)

However, when I use the same contact in Anura3D_2019. It can calculate successfully. It confused me so much!
Is there some problem with the version of Anura3D 2022 on my computer??

**Comment by FrancescaCeccato** [comment]
What is the name of the material that constitutes the soil? in the contact specification you have to assign the contact to the foundation and the "name" must be the name of the material in contact with your foundation, for example "soil". I can see from the picture that you wrote "footing", I don't think it is correct.
Furthermore, please use only 2 materials and material 1 must be the soil.
I hope this fix the issue.

**Reply by Tiancheng-Xie** [reply]
Dear Prof. Ceccato, thanks for your replying and help! Follow your advice, I have tried again. But the same error always occurs. Please see the pictures. Thanks for your time.
![image](https://user-images.githubusercontent.com/106669043/222302362-ce95df8f-a0ac-42f3-b69f-eb039b57be76.png)
(as you mentioned, only two materials are used and the soil is first assigned.)
![image](https://user-images.githubusercontent.com/106669043/222302632-8a75ea48-d587-4d7d-86a9-c79a2b67f7e4.png)
(boundary conditions)
![image](https://user-images.githubusercontent.com/106669043/222302754-b5b5b8e3-c8e7-4f98-b678-86a7d4fa15ef.png)
(Load conditions)
![image](https://user-images.githubusercontent.com/106669043/222302848-abc568ca-d82e-498c-8661-64650b779ba0.png)
(Contact properties)
![image](https://user-images.githubusercontent.com/106669043/222302971-e245aba3-8a44-4735-aaa5-d0279de913c7.png)
(Calculation data)
![image](https://user-images.githubusercontent.com/106669043/222303238-c70d93fd-7a73-42fa-bf01-6e4b1248f469.png)
(error)

**Comment by albuquerqueraniel** [comment]
Hello, Tiancheng-Xie.

I've faced this same problem in a simulation I'm working on. I don't know why, but Anura is trying to assign the load value to the second index of EXTLOADTOTAL, which doesn't exist when you have defined only one load system. If you define two loads: one in system A and another in system B it might be solved (they don't need to be active, only defined). 

Best regards,
Raniel

**Reply by Tiancheng-Xie** [reply]
Dear Raniel,

Perfect! your suggestion is very useful! Thank you so much for all your help. l appreciate it!

**Comment by FrancescaCeccato** [comment, marked as answer]
There was a bug in the code, which i solved now. Thank you for pointing this out.

**Reply by Tiancheng-Xie** [reply]
Dear Porf. Ceccato,

Thanks for your time! Shold I download the newest code to solve the bug?

**Reply by FrancescaCeccato** [reply]
yes, can you try to see if that solved your issue and let me know?
